### PR TITLE
Mapped keyboard keys in EbookReader Activity

### PR DESCRIPTION
**Pull request by NischayGoyal1**
Hey @llaske , I have mapped the left and right keyboard keys to Pagination , Now we can change pages just by pressing keys
Please Review it
Issue : https://github.com/llaske/sugarizer/issues/1273

## Video

https://user-images.githubusercontent.com/81116984/223002252-aa62b63c-ca9a-4e97-b071-73cd85ff1dde.mp4

### CODE REVIEW

**Comment by llaske** [comment]
Not a good idea to implement it like that. The listener is added to the whole document so there is an error when back to the library and pressing keys.

image

Plus the mounted method is call each time the ebook is displayed so the listener is added multiple times.

**Comment by llaske** [comment]
It's not good, add a trace in your code, you will see that the listener is added each time a book is open.

**Comment by NischayGoyal1** [comment]
#### I have implemented a fix by adding a "beforedestroy" hook. Specifically, I have added a removeEventListener function to the "beforedestroy" hook, which removes the event listeners whenever the Ebook component is destroyed. This ensures that the event listeners are only added once when the Ebook component is mounted and are removed when the component is destroyed, preventing the duplication of event listeners.
#### the reason why I added the event listeners in a separate JavaScript file was because the Ebook component was being mounted separately in a different file. As a result, there was no other way to add the event listeners without modifying that file.

I have made the required changes please review it

File: activities/EbookReader.activity/js/reader.js [hunk 1]
```diff
@@ -11,6 +11,20 @@ var EbookReader = {
...
+			vm.previousPage();
+		  } else if (event.keyCode == 39) {
+			vm.nextPage();
+		  }
+		});
+	  },
+	  beforeDestroy: function() {
+		// Remove the event listener
+		document.removeEventListener("keydown", this.handleKeyDown);
+	  },
```
Comment (llaske) [inline activities/EbookReader.activity/js/reader.js:27]: Look like indentation is wrong here.

Comment (NischayGoyal1) [reply activities/EbookReader.activity/js/reader.js:27]: Fixed it

**Comment by llaske** [comment]
Nice. Thanks.

**Pull request merged**